src/Experiment/src/ModuleBuilder.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Reflection.Emit.Experimental
+{
+    public class ModuleBuilder : System.Reflection.Module
+    {
+        private readonly Guid _s_guid;
+        public override System.Reflection.Assembly Assembly { get; }
+        public override string ScopeName { get; }
+
+        internal ModuleBuilder(string name, Assembly assembly)
+        {
+         //Example random GUID - I still need to understand when, how and why to generate them.
+        _s_guid= Guid.NewGuid();
```

Comment:
It should not be necessary to create the random Guid upfront and store it. You can create just when you needed, like:

`metadataBuilder.GetOrAddGuid(Guid.NewGuid())`